### DISCUSSION TITLE
Publish on CRAN?

### DISCUSSION THREAD

**Discussion by enricoschumann** [lbilli/rib#16] [General]
I have said it before: it's good to see an up-to-date pure-R implementation of the IB API.

Just out of curiosity: have you considered publishing your package on CRAN?

**Comment by lbilli** [comment]
It has not been a top priority of mine but I might take a look at what's needed in order to be accepted by CRAN.

**Comment by enricoschumann** [comment]
I'd encourage you to release on CRAN.  I think that a majority of R users only uses packages from CRAN (via install.packages() ), and so won't use your package.

The CRAN requirements are not that high -- mainly documentation, a clean list of dependencies (in the DESCRIPTION file), and a willingness to fix issues that they see in their test infrastructure (e. g. when there were changes in base R after a new release).  The latter in particular is a very beneficial feature, I think.

Essentially, you'll see what is missing by running

     `R CMD check --as-cran rib_0.17.2.tar.gz`

on the tarball (from `R CMD build`).

**Comment by lbilli** [comment]
It's now available on [CRAN](https://cran.r-project.org/package=rib).

**Comment by riazarbi** [comment]
Congratulations! I just checked back on your repo after about 6 months and my timing is impeccable. Well done, I know the CRAN hoops can be quite challenging.

**Comment by enricoschumann** [comment]
Great!

(Your package is now also listed in the Finance Task View, [`https://github.com/cran-task-views/Finance/blob/master/Finance.md`](https://github.com/cran-task-views/Finance/blob/master/Finance.md), to be mirrored on CRAN [https://cran.r-project.org/web/views/Finance.html](https://cran.r-project.org/web/views/Finance.html).)
